soc/arm/nxp_kinetis/k6x/soc.c

```diff
@@ -111,9 +112,10 @@ static ALWAYS_INLINE void clock_init(void)
 #if CONFIG_ETH_MCUX_RMII_EXT_CLK
 	CLOCK_SetRmii0Clock(1);
 #endif
-#if CONFIG_USB_KINETIS || CONFIG_UDC_KINETIS
+#if CONFIG_USB_KINETIS || CONFIG_UDC_KINETIS || CONFIG_USB_UHC_NXP_KHCI
 	CLOCK_EnableUsbfs0Clock(kCLOCK_UsbSrcPll0,
 				DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency));
+	SYSMPU_Enable(SYSMPU, 0);
```

Comment:
Might want a comment here highlighting that the SYSMPU must be enabled for USB to function. I would also only call this if `CONFIG_ARM_MPU` is not set.